.github/workflows/track-prs-for-release.yml

```diff
@@ -5,10 +5,57 @@ on:  # yamllint disable-line rule:truthy
   pull_request_target:
     types:
       - closed
+      - labeled
+      - unlabeled
+
+# All jobs require a specifically generated token to run, since we want to
+# access org-level projects and the normal GITHUB_TOKEN only has access to this
+# repository.
 
 jobs:
-  add_merged_pr_to_release_project:
-    if: github.base_ref == 'main' && github.event.pull_request.merged
+  handle_merged_pr:
+    if: github.base_ref == 'main' && github.event.action == 'closed' && github.event.pull_request.merged
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.base_ref }}
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92
+        with:
+          app_id: ${{ secrets.RELEASE_APP_ID }}
+          private_key: ${{ secrets.RELEASE_APP_PRIVATE_KEY }}
+      - name: Handle merged PR
+        env:
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
+          PR_ID: ${{ github.event.pull_request.node_id }}
+        run: tools/scripts/track-pr pr-merged "$PR_ID"
+
+  handle_labeled_pr:
+    if: github.base_ref == 'main' && github.event.action == 'labeled'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          ref: ${{ github.base_ref }}
+      - name: Generate token
+        id: generate_token
+        uses: tibdex/github-app-token@b62528385c34dbc9f38e5f4225ac829252d1ea92
+        with:
+          app_id: ${{ secrets.RELEASE_APP_ID }}
+          private_key: ${{ secrets.RELEASE_APP_PRIVATE_KEY }}
+      - name: Handle labeled PR
+        env:
+          GITHUB_TOKEN: ${{ steps.generate_token.outputs.token }}
+          PR_ID: ${{ github.event.pull_request.node_id }}
+          PR_LABEL: ${{ github.event.label.name }}
+        run: tools/scripts/track-pr pr-labeled "$PR_ID" "$PR_LABEL"
```

Comment:
The labeled and unlabeled steps could probably be merged together into one.  The argument `pr-labeled` is also `pr-${{ github.event.action }}`, so updating the job's conditional to  `if` just checked action == labeled || action == unlabeled` should allow some reduced duplication of code, right? :)